extension/background.js

```diff
@@ -106,28 +106,40 @@ class Background {
   async getAlertBadge_(tab) {
     const alertList = await alerts.computeAlerts(tab.url);
     this.alerts = alertList;
-    if (alertList.length > 0) {
-      chrome.browserAction.setBadgeText({
-        text: alertList.length.toString(),
-        tabId: tab.id,
-      });
+
+    /**
+     * Sets the color of the flag icon.
+     * @param {!string} color The color to set the flag. Will be one of orange,
+     *     green, or gray.
+     * @param {!Tab} tab A Chrome Tab instance.
+     */
+    const setFlagIconColor = (color, tab) => {
       chrome.browserAction.setIcon({
         path: {
-          '16': 'images/orangeflag16.png',
-          '48': 'images/orangeflag48.png',
-          '128': 'images/orangeflag128.png',
+          '16': `images/${color}flag16.png`,
+          '48': `images/${color}flag48.png`,
+          '128': `images/${color}flag128.png`,
         },
         tabId: tab.id,
       });
+    };
+
```

Comment:
optional nit: I might rewrite the below code with an early-return to reduce nesting:

```
if (alertList.length == 0) {
  setFlagIconColor('green', tab);
  return;
}
if (alertList.length == 1 && alertList.includes(...)) {
  setFlagIconColor('gray', tab);
} else {
...
  setFlagIconColor('orange', tab);
}
```

(this is just Chromium style so up to you if you want to follow it here!)

Reply:
I like that styling--updated. Thanks!